Assets/Scripts/DayNightCycle/DayAndNightControl.cs

```diff
@@ -0,0 +1,246 @@
+﻿using System;
+using System.Collections;
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.Animations;
+using UnityEditor;
+using Assets.Scripts.Services;
+using Assets.Scripts.Settings;
+
+namespace Assets.Scripts.DayNightCycle
+{
+    public class DayAndNightControl : MonoBehaviour
+    {
+        public GameObject moon;
+        public ReflectionProbe reflectionProbe;
+        public float sunTilt = -15f;
+        [SerializeField] private TimecycleProfile timecycle = null;
+        [SerializeField] private float sunRotationOffset = 0f;
+        [Tooltip("The amount of frames to wait before doing the next lighting update")]
+        [SerializeField] private int lightingUpdateInterval = 10;
+        public Material skyBoxPROCEDURAL;
+
+        [Range(0f, 24f)] public float currentTime;
+        public float CurrentTime01 => currentTime / 24;
+        public Slider TimeSlider  = null;
+        public Camera MoonCamera = null;
+        public Camera MainCamera = null;
+        public int currentDay = 0;
+        public Light directionalLight;
+        public float DayLength = 300f; //default value is 300 seconds in one day
+        public bool pause { get; set; }
+        public float lightIntensity; //static variable to see what the current light's insensity is in the inspector
+
+        Camera targetCam;
+        private int frames;
+       
+        // Use this for initialization
+        void Start()
+        {
+            pause=true;
+            Service.Settings.OnTimeSettingsChanged += Settings_OnTimeSettingsChanged;
+            MoonCamera = GetComponentInChildren<Camera>();
+
+            foreach (Camera c in GameObject.FindObjectsOfType<Camera>())
+            {
+                if (c.isActiveAndEnabled)
+                {
+                    targetCam = c;
+                }
+            }
+            lightIntensity = directionalLight.intensity; // What's the current intensity of the sunlight
+
+            UpdateLightingSettings();
+            UpdateLight(); // Initial lighting update.
+
+            IEnumerator SetupCameraConstraintAndReflectionProbe()
```

Comment:
Local Functions are always at the bottom of the method

Reply:
@WattanaGaming  I think you added this would you like to take it up?

Reply:
That was my habit of putting local functions right above where they're used. Will fix